examples/lightningsynth.py

```diff
@@ -63,6 +65,20 @@ def __len__(self):
         return self.num_batches
 
 
+class TorchSynthCallback(pl.Callback):
```

Comment:
Yeah this is cool, so our API could be that you pass in an optional function, and that function does whatever it wants with this batch. That function also gets the batch index as a param

Reply:
We can do that later